src/web/components/Hydrate.tsx

```diff
@@ -0,0 +1,20 @@
+import ReactDOM from 'react-dom';
+
+type Props = {
+    root: IslandType;
+    children: JSX.Element;
+};
+
+export const Hydrate = ({ root, children }: Props) => {
+    const element = document.getElementById(root);
+    if (!element) return null;
+    window.performance.mark(`${root}-hydrate-start`);
+    ReactDOM.hydrate(children, element);
+    window.performance.mark(`${root}-hydrate-end`);
+    window.performance.measure(
+        `${root}-hydrate`,
+        `${root}-hydrate-start`,
+        `${root}-hydrate-end`,
+    );
+    return null;
```

Comment:
do you even need to return anything?

Reply:
You do. It's a react component so it needs to return either an Element or null (plus typescript will shout at you if you don't)